csp/impl/types/common_definitions.py

```diff
@@ -53,7 +53,11 @@ def __new__(cls, *args, **kwargs):
             kwargs = {k: v if not isTsBasket(v) else OutputBasket(v) for k, v in kwargs.items()}
 
         # stash for convenience later
-        kwargs["__annotations__"] = kwargs
+        kwargs["__annotations__"] = kwargs.copy()
+        try:
```

Comment:
Both are possible but this block doesn't change behavior to end users and is closer to the end state: i.e. where csp requires pydantic>2 and this code is executed all the time. The advantage to having it like this is that all the existing unit tests will check the code in ci/cd because pydantic is listed as a dev dependency. By making it depend on the env variable, a whole separate set of tests is needed, only for them to be deleted in the next step.